iceoryx_hoofs/concurrent/buffer/include/iox/detail/spsc_fifo.inl

```diff
@@ -31,27 +31,69 @@ inline bool SpscFifo<ValueType, Capacity>::push(const ValueType& value) noexcept
     {
         return false;
     }
-    auto currentWritePos = m_write_pos.load(std::memory_order_relaxed);
+    // Memory order relaxed is enough since:
+    // - there is no concurrent access to this method
+    // - the load statement cannot be reordered with writing m_data otherwise it would not compile
+    auto currentWritePos = m_writePos.load(std::memory_order_relaxed);
     m_data[currentWritePos % Capacity] = value;
 
-    // m_write_pos must be increased after writing the new value otherwise
-    // it is possible that the value is read by pop while it is written.
-    // this fifo is a single producer, single consumer fifo therefore
-    // store is allowed.
-    m_write_pos.store(currentWritePos + 1, std::memory_order_release);
+    // SYNC POINT WRITE: m_data
+    // We need to make sure that writing the value happens before incrementing the
+    // m_writePos otherwise the following scenario can happen:
+    // 1. m_writePos is increased (but the value has not been written yet)
+    // 2. Another thread calls pop(): we check if the queue is empty => no
+    // 3. In pop(), when we read a value, a data race can occur when at the same time a value is
+    // written by push. With memory_order_release, this cannot happen as it is guaranteed that
+    // writing the data happens before incrementing m_writePos. Note that the following scenario
+    // can still happen (but this is not a problem):
```

Comment:
Indeed. This is an inherent race with all concurrent algorithms. It depends on when you look at something and in a blink of an eye this information might be outdated.

Reply:
Oh, could you change this to ` but, although it is an inherent race with concurrent algorithms, it is not a data race and therefore not a problem.` to make it explicit why it is not a problem.